### PR TITLE
int blends, some simd, all around faster

### PR DESCRIPTION
**Pull request by guzba**
before:

```
draw bmNormal ....................... 6.953 ms    ±0.377  x10
draw Smooth bmNormal ............... 16.723 ms    ±0.142  x10

run_files.nim Driving - Navigation: 3.28490686416626
svg decode ........................ 285.880 ms    ±3.089  x10
```

after:

```
draw bmNormal ....................... 4.625 ms    ±0.319  x10
draw Smooth bmNormal ............... 15.330 ms    ±0.228  x10

run_files.nim Driving - Navigation: 2.821513175964356
svg decode ........................ 235.707 ms    ±1.834  x10
```

### CODE REVIEW

**Comment by guzba** [comment]
```
blendNormal ......................... 5.595 ms    ±0.113  x10
blendNormalFloats .................. 17.934 ms    ±0.071  x10
blendDarken ......................... 7.807 ms    ±0.223  x10
blendDarkenFloats .................. 30.119 ms    ±0.119  x10
blendMultiply ....................... 7.496 ms    ±0.289  x10
blendMultiplyFloats ................ 29.610 ms    ±0.169  x10
blendLinearBurn ..................... 7.839 ms    ±0.239  x10
blendLinearBurnFloats .............. 29.810 ms    ±0.168  x10
blendColorBurn ..................... 14.482 ms    ±0.164  x10
blendColorBurnFloats ............... 27.079 ms    ±0.104  x10
blendLighten ........................ 7.817 ms    ±0.213  x10
blendLightenFloats ................. 29.973 ms    ±0.087  x10
blendScreen ......................... 7.453 ms    ±0.249  x10
blendScreenFloats .................. 30.946 ms    ±0.207  x10
blendLinearDodge .................... 7.420 ms    ±0.225  x10
blendLinearDodgeFloats ............. 29.500 ms    ±0.129  x10
blendColorDodge .................... 14.358 ms    ±0.141  x10
blendColorDodgeFloats .............. 33.459 ms    ±0.886  x10
blendOverlay ........................ 7.781 ms    ±0.403  x10
blendOverlayFloats ................. 28.561 ms    ±0.348  x10
blendSoftLight ...................... 4.042 ms    ±0.058  x10
blendSoftLightFloats ............... 33.186 ms    ±0.373  x10
blendHardLight ...................... 7.488 ms    ±0.256  x10
blendHardLightFloats ............... 27.753 ms    ±0.286  x10
blendDifference ..................... 7.829 ms    ±0.212  x10
blendDifferenceFloats .............. 25.506 ms    ±0.112  x10
blendExclusion ...................... 7.437 ms    ±0.201  x10
blendExclusionFloats ............... 30.530 ms    ±0.168  x10
blendHue ........................... 38.105 ms    ±0.236  x10
blendHueFloats ..................... 38.907 ms    ±0.183  x10
blendSaturation .................... 39.074 ms    ±0.207  x10
blendSaturationFloats .............. 39.099 ms    ±0.178  x10
blendColor ......................... 27.228 ms    ±0.158  x10
blendColorFloats ................... 29.247 ms    ±0.234  x10
blendLuminosity .................... 37.434 ms    ±0.465  x10
blendLuminosityFloats .............. 36.854 ms    ±0.108  x10
blendMask ........................... 1.435 ms    ±0.006  x10
blendMaskFloats ..................... 9.388 ms    ±0.269  x10
blendSubtractMask ................... 1.446 ms    ±0.040  x10
blendSubtractMaskFloats ............. 1.431 ms    ±0.001  x10
blendIntersectMask .................. 1.432 ms    ±0.004  x10
blendIntersectMaskFloats ............ 1.432 ms    ±0.003  x10
blendExcludeMask .................... 1.437 ms    ±0.011  x10
blendExcludeMaskFloats .............. 8.553 ms    ±0.265  x10
```

File: src/pixie/images.nim [hunk 1]
```diff
@@ -460,7 +460,7 @@ proc drawUber(
           if smooth:
             b.getRgbaSmooth(xFloat, yFloat)
           else:
-            b.getRgbaUnsafe(xFloat.round.int, yFloat.round.int)
+            b.getRgbaUnsafe(xFloat.int, yFloat.int)
```
Comment (treeform) [inline src/pixie/images.nim:463]: I think the rounding here is very important and it introduces tiny artifacts when not rounding. See the image_9_patch for rounding requirement.

File: src/pixie/images.nim [hunk 2]
```diff
@@ -406,17 +402,24 @@ proc drawCorrect*(a, b: Image, mat: Mat3, blendMode: BlendMode) =
     minFilterBy2 /= 2
     matInv = matInv * scale(vec2(0.5, 0.5))
 
+  let smooth = not(dx.length == 1.0 and dy.length == 1.0 and
```
Comment (treeform) [inline src/pixie/images.nim:405]: I would like to keep draw correct free of "fancy" logic. 

Could you please create a different version of draw correct with the vector motion if you want to test it, but keep draw correct as simple and plain as possible?

File: src/pixie/images.nim (hunk 1 above)
Comment (guzba) [reply src/pixie/images.nim]: ah ok, I put it back

File: src/pixie/images.nim (hunk 2 above)
Comment (guzba) [reply src/pixie/images.nim]: I removed it, not important to me was just comparing

**Pull request merged**